lib/src/common/repos_service.dart

```diff
@@ -134,69 +153,89 @@ class RepositoriesService extends Service {
+    return _github.postJSON(
+      "/repos/${repo.fullName}",
+      body: jsonEncode(data),
+      statusCode: 200,
+    );
   }
 
   /// Deletes a repository.
   ///
   /// Returns true if it was successfully deleted.
   ///
   /// API docs: https://developer.github.com/v3/repos/#delete-a-repository
-  Future<bool> deleteRepository(RepositorySlug slug) {
-    return _github
-        .request('DELETE', '/repos/${slug.fullName}')
-        .then((response) => response.statusCode == StatusCodes.NO_CONTENT);
+  Future<void> deleteRepository(RepositorySlug slug) async {
+    await _github.request(
+      'DELETE',
+      '/repos/${slug.fullName}',
+      statusCode: StatusCodes.NO_CONTENT,
+    );
   }
 
   /// Lists the contributors of the specified repository.
   ///
   /// API docs: https://developer.github.com/v3/repos/#list-contributors
   Stream<User> listContributors(RepositorySlug slug, {bool anon = false}) {
-    return PaginationHelper(_github).objects(
-        'GET', '/repos/${slug.fullName}/contributors', User.fromJson,
-        params: {"anon": anon.toString()});
+    return PaginationHelper(_github).objects<Map<String, dynamic>, User>(
+      'GET',
+      '/repos/${slug.fullName}/contributors',
+      (i) => User.fromJson(i),
+      params: {"anon": anon.toString()},
+    );
   }
 
   /// Lists the teams of the specified repository.
   ///
   /// API docs: https://developer.github.com/v3/repos/#list-teams
   Stream<Team> listTeams(RepositorySlug slug) {
-    return PaginationHelper(_github)
-        .objects('GET', '/repos/${slug.fullName}/teams', Team.fromJSON);
+    return PaginationHelper(_github).objects<Map<String, dynamic>, Team>(
+      'GET',
+      '/repos/${slug.fullName}/teams',
+      (i) => Team.fromJSON(i),
+    );
   }
 
   /// Gets a language breakdown for the specified repository.
   ///
   /// API docs: https://developer.github.com/v3/repos/#list-languages
-  Future<LanguageBreakdown> listLanguages(RepositorySlug slug) =>
-      _github.getJSON("/repos/${slug.fullName}/languages",
-          statusCode: StatusCodes.OK,
-          convert: (Map<String, dynamic> input) =>
-              LanguageBreakdown(input.cast<String, int>()));
+  Future<LanguageBreakdown> listLanguages(RepositorySlug slug) async =>
+      _github.getJSON<Map<String, dynamic>, LanguageBreakdown>(
+        "/repos/${slug.fullName}/languages",
+        statusCode: StatusCodes.OK,
+        convert: (input) => LanguageBreakdown(input.cast<String, int>()),
+      );
 
   /// Lists the tags of the specified repository.
   ///
   /// API docs: https://developer.github.com/v3/repos/#list-tags
   Stream<Tag> listTags(RepositorySlug slug) {
-    return PaginationHelper(_github)
-        .objects('GET', '/repos/${slug.fullName}/tags', (j) => Tag.fromJson(j));
+    return PaginationHelper(_github).objects<Map<String, dynamic>, Tag>(
+      'GET',
+      '/repos/${slug.fullName}/tags',
+      (j) => Tag.fromJson(j),
```

Comment:
#nit everything else uses `i` and this one uses `j` .. 🤷‍♂ could be `i` for consistency.
```suggestion
      (i) => Tag.fromJson(i),
```